grails-app/services/au/org/ala/specieslist/HelperService.groovy

```diff
@@ -596,6 +685,27 @@ class HelperService {
         }
     }
 
+    def  matchValuesToSpeciesListItem(String[] values, Map termIndex, SpeciesListItem sli){
+        String rawScientificName = termIndex.containsKey(RAW_SCIENTIFIC_NAME) ? values[termIndex[RAW_SCIENTIFIC_NAME]] : null
```

Comment:
Perhaps make this a closure? def get = { key -> termIndex.containsKey[key] ? values[termIndex[key] : null } and then String family = get(FAMILY)

Reply:
closure looks smart but hard to debug.